pkgs/development/libraries/amtk/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv, autoconf, autoconf-archive, automake, intltool, gnome2, pkgconfig
+, libtool, gobjectIntrospection, gnome3 }:
+let
+  version = "4.99.1";
+in stdenv.mkDerivation {
+  name = "amtk-${version}";
+  src = fetchTarball {
+    url = "https://gitlab.gnome.org/GNOME/amtk/-/archive/${version}/amtk-${version}.tar.bz2";
+    sha256 = "0d7da1r4xxh74zrdawy4cgr4pvbdfrjs9vp7r2jsbv3xscxqrmiq";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    autoconf-archive
+    gobjectIntrospection
+    automake
+    intltool
+    gnome2.gtkdoc
+    pkgconfig
+    libtool
+    gnome3.gtk.dev
+  ];
+
+  configurePhase = ''
+    mkdir -p $out
+    ./autogen.sh --prefix=$out
```

Comment:
Usually, you would use `preConfigure = ''NOCONFIGURE=1 ./autogen.sh'';` and let the generic builder take care of the rest.